README.md
# rAntenna
rAntenna は Ruby で記述された Web サイトの更新時刻取得エージェントです。 以下の特徴があります。

* Ruby で書かれている。
* 出力形式が RDF (HTML 出力も対応)

## ダウンロード
リリース版は以下からダウンロード可能です。

**最新版**

* https://github.com/downloads/elpeo/rantenna/rantenna-0.0.6.tar.gz [README](http://elpeo.jp/diary/20060113p01.html)

**安定版**

* https://github.com/downloads/elpeo/rantenna/rantenna-0.0.5.tar.gz [README](http://elpeo.jp/diary/20040629p01.html)

**古いバージョン**

* https://github.com/downloads/elpeo/rantenna/rantenna-0.0.4.tar.gz [README](http://elpeo.jp/diary/20040423p01.html)
* https://github.com/downloads/elpeo/rantenna/rantenna-0.0.3.tar.gz [README](http://elpeo.jp/diary/20040416p01.html)

プログラム本体の他に、

* Ruby1.6.7 以降
* [uconv](http://raa.ruby-lang.org/project/uconv) (Ruby1.8.1 以前で UTF-8 で出力する場合)
* [ruby-zlib](http://raa.ruby-lang.org/project/ruby-zlib) (Ruby1.6 で input_lirs.rb, output_lirs.rb を使う場合)

が必要です。

## 動作例
http://elpeo.jp/antenna/ で rAntenna を利用した更新時刻取得サイトを運用しています。

## 設定
### インストール

rAntenna は Ruby で書かれていますので、 まず Ruby がインストールされているか確認します。 インストールされてなかった場合はまず Ruby をインストールしてください。

次にダウンロードした rAntenna を適当な場所で以下のように解凍します。

    $ tar xvzf rantenna.tar.gz

解凍が完了すると、rantenna と言うディレクトリができています。 このディレクトリを /var/www/html/rantenna 等、 Web サーバーの公開ディレクトリにコピーすればインストールは完了です。

### .htaccess の設定
配布ファイルの中に dot.htaccess というファイルが同梱されていますので、 こちらを .htaccess にコピーします。

### antenna.conf の設定
rAntenna の設定ファイルは antenna.conf です。 配布ファイルの中には antenna.conf.sample というサンプル設定ファイルが 同梱されていますので、こちらを antenna.conf にコピーし、 必要な変更します。 必須項目は以下のとおりです。

```
@title       アンテナのタイトルを入れてください。
@copyright   Copyright を入れてください。
@antenna_url アンテナのURLを入れてください。
@rdf_url     出力されるRDFファイルのURLを入れてください。
@urls        巡回するURLを入れてください。
             形式は、
             タイトル, 作者, URL, 巡回URL
             です。巡回URLが無い場合はURLを使います。
```

以下の項目は任意で設定してください。

```
@filters     フィルタを入れてください。
@rdf_path    出力されるRDFファイル名を入れてください。
             絶対パスも使用できます。
@link_format URLからリンクを作成するフォーマットを指定してください。
             詳しい記法は antenna.conf.sample を参照ください。
```

@filters の設定方法については、フィルタをご参照ください。

@link_format で go.rb を使う場合の記述方法については、 [go.rb](http://elpeo.jp/wiki/?go.rb) をご参照ください。

## crontab の設定
rAntenna では update.rb というコマンドを実行したときに Web サイトの更新時刻を取得します。

定期的に更新時刻を取得する場合は cron を使い、 定期的に update.rb が実行されるようにします。

update.rb を実行する際の引数として、conf ファイルのパスを１つまたは複数指定できます。引数を指定しない場合は、update.rb と同じディレクトリにある antenna.conf というファイルを使用します。

## 拡張
この他の機能拡張・利用法に関しては [rAntenna plugin](http://elpeo.jp/wiki/?rAntenna+plugin) を参照ください。

* LIRS 入力
* HTML 出力
* LIRS 出力

などの話題を扱っています。


ChangeLog
Mon Aug 21 11:22:48 2006  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: support relative redirect.

Sat Aug 19 21:51:48 2006  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: use IO::foreach.

Fri Aug 18 15:02:14 2006  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* fix plural (include|exclude)_range bug.

Wed Jan 11 14:49:12 2006  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* release 0.0.6.

Wed Jan  4 19:38:46 2006  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb, antenna.conf.sample: support filter.

Tue Jan  3 18:36:47 2006  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: fix last year's date bug.

Thu Dec 22 16:17:09 2005  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* misc/plugin/output_imode.rb: support sharp button.

Thu Dec 22 00:10:10 2005  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb, dot.htaccess, misc/plugin/output_imode.rb: support i-mode.

Wed Dec 21 12:55:54 2005  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb, update.rb, misc/plugin/output_html.rb,
	misc/plugin/output_lirs.rb: support @confdir.
	* antenna.rb: support MD5 check and REDIRECT.

Mon May  9 17:36:00 2005  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* index.xsl, misc/skel/template.html: remove <TT> tag.
	* theme/rantenna/rantenna.css: add font-family.

Wed Oct  6 16:20:26 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* index.xsl, misc/skel/template.html: add RSS meta tag.

Mon Jun 28 18:09:23 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* release 0.0.5.

Sun Jun 27 00:00:56 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* go.rb, update.rb: add BEGIN { $defout.binmode }.

Fri Jun 25 22:59:37 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: ignore tags and spaces when get_last_modified.

Fri Jun 18 15:11:47 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb, index.xsl: add timezone at dc:date.

Thu Jun 17 23:50:32 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: add add_*_proc and antenna_url method.
	* update.rb, go.rb: use antenna_url method.

Mon May 17 01:42:30 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: support WWWC type meta tag.

Thu Apr 29 13:26:16 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: use latest date when cannot get last-modified.

Thu Apr 22 19:10:26 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: add urls_compare method.

Tue Apr 20 19:03:09 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* misc/skel/simple_template.html: move from misc/template.html.sample
	* misc/skel/template.html: add.
	* theme/rantenna/rantenna.css: add adminmenu.

Tue Apr 20 17:37:49 2004  Norihiro Hattori  <tnh@webmasters.gr.jp>

	* theme/rantenna/rantenna.css: change design.

Tue Apr 20 13:11:04 2004  Norihiro Hattori  <tnh@webmasters.gr.jp>

	* index.xsl: change css from default.css to rantenna.css.
	* misc/template.html.sample: change css from default.css to rantenna.css.
	* theme/*: added from tDiary.

Tue Apr 20 10:43:21 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: use CGI::escapeHTML.

Mon Apr 19 18:52:49 2004  Norihiro Hattori  <tnh@webmasters.gr.jp>

	* misc/template.html.sample: add css

Fri Mar 26 12:23:43 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* plugin/README: add.

Thu Mar 25 23:31:52 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: add @generator.
	* misc/template.html.sample: add.

Thu Mar 18 13:01:33 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* go.rb: support http:/ type Apache bug?

Thu Mar 18 11:08:00 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: return link unless @link_format.

Thu Mar 18 10:30:17 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: support @link_format and delete @use_suffix.

Wed Mar 17 18:52:54 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* go.rb: add.

Tue Mar 16 17:21:49 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* TODO: add.

Fri Mar 12 19:53:57 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: fix set previous update time bug.

Fri Mar 12 11:52:53 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* fix bug cannot read date from rdf if use suffix.

Fri Mar 12 11:15:37 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: delete tags if parse response.body.

Thu Mar 11 22:23:20 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: year-- if month > now.month

Thu Mar 11 19:53:21 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: support Last-Modified format.

Thu Mar 11 18:27:47 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: support Lastupdate format.

Thu Mar 11 17:59:08 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* update.rb: use instance_eval().

Thu Mar 11 16:33:00 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: support 2-lines date format.

Thu Mar 11 15:34:37 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb, update.rb: add get access and delete touch function.

Tue Mar  9 10:47:43 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: delete description.

Sat Mar  6 01:47:17 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: use File.expand_path().

Sat Mar  6 00:19:38 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: unset @last_detected unless @last_modified.

Fri Mar  5 20:35:04 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: use @dir if current dir.
	* antenna.conf.sample: add comment.

Fri Mar  5 19:54:41 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* antenna.rb: Add suffix mode.

Fri Mar  5 19:01:58 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* *.rb: use File.join() instead of '/'

Fri Mar  5 15:12:16 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* index.xsl: bug fixed (change dc:creator to dc:publisher).

Fri Mar  5 12:43:32 2004  Norihiro Hattori  <tnh@webmasters.gr.jp>

	* index.xsl: bug fixed (change dc:publisher to dc:creator).

Thu Mar  4 23:29:55 2004  Michitaka Ohno  <elpeo@mars.dti.ne.jp>

	* index.xsl: Add DOCTYPE


update.rb
#!/usr/bin/env ruby
#
# update.rb $Revision: 1.4 $
#
# Copyright (C) 2004  Michitaka Ohno <elpeo@mars.dti.ne.jp>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.

BEGIN { $defout.binmode }

require File.expand_path( 'antenna.rb', File::dirname( __FILE__ ) )

if ENV['REQUEST_METHOD'] then
	require 'cgi'
	path = Antenna.new.antenna_url
	print CGI.new.header( {'type' => 'text/html'} )
	print <<-HTML
	<html>
	<head>
	<meta http-equiv="refresh" content="0;url=#{path}">
	<title>moving...</title>
	</head>
	<body>Wait or <a href="#{path}">Click here!</a></body>
	</html>
	HTML
elsif ARGV.empty? then
	a = Antenna.new
	a.go_round
	a.output
else
	ARGV.each do |item|
		a = Antenna.new( item )
		a.go_round
		a.output
	end
end



misc/plugin/input_lirs.rb
# input_lirs.rb $Revision: 1.5 $
#
# Copyright (C) 2004  Michitaka Ohno <elpeo@mars.dti.ne.jp>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.

# Last modified Information Relaying Specification ver.2.1
# http://aniki.haun.org/natsu/natsu3.1b/doc/LIRS.html

# antenna.conf
# @input_lirs_urls = [
# 'http://natsu.tnh.jp/natsumican.lirs',
# ]

require 'net/http'
require 'timeout'

eval( <<-TOPLEVEL_CLASS, TOPLEVEL_BINDING )
class String
	def csplit
		r = []
		s = ""
		each_byte do |byte|
			if byte == ?, && s[-1] != ?\\\\ then
				r << s
				s = ""
			else
				s << byte
			end
		end
		r << s
	end
end
TOPLEVEL_CLASS

class StringReader
	def initialize( s )
		@str = s
			@index = 0
	end
	def read( length = nil )
		if @index < @str.length then
			nth = @index
			len = length || @str.lenth
			@index += len
			@str[nth, len]
		else
			length ? nil : ""
		end
	end
end

def input_lirs
	lirs_urls = @input_lirs_urls||[]

	limittime = 10

	lirs_urls.each do |url|
		next unless %r[^http://([^/]+)(/.*)?$] =~ url
		path = $2 || '/'
		host, port = $1.split( /:/ )
		port = '80' unless /^[0-9]+$/ =~ port
		timeout( limittime ) do
			begin
				Net::HTTP.version_1_1
				Net::HTTP.start( host.untaint, port.to_i.untaint ) do |http|
					response, = http.get( path )
					buf = response.body
					if /gzip/i =~ response['Content-Encoding'] || /.gz$/i =~ url then
						begin
							require 'zlib'
							sr = StringReader.new( buf )
							gz = Zlib::GzipReader.new( sr )
							buf = gz.read
							gz.close
						rescue LoadError
						end
					end
					buf.each do |line|
#						lirs = line.strip.split( /\,/ )
						lirs = line.strip.csplit
						next unless lirs[0] == 'LIRS'
						lm = lirs[1].to_i
						ld = lirs[2].to_i
						cl = lirs[4].to_i
						site = lirs[5]
						au = lirs[8]
						next if lm == 0
						date = Time.at( lm )
						next if @last_modified[site] && @last_modified[site] > date
						@last_modified[site] = date
						@last_detected[site] =  ld == 0 ? nil : Time.at( ld )
						@content_length[site] = cl
						if au && au != "" && au != "0" then
							@auth_url[site] = au
						else
							@auth_url[site] = File.dirname( url ) + '/'
						end
					end
				end
			rescue Exception
			rescue
			end
		end
	end
end

add_input_proc do
	input_lirs
end


misc/plugin/output_html.rb
# output_html.rb $Revision: 1.6 $
#
# Copyright (C) 2004  Michitaka Ohno <elpeo@mars.dti.ne.jp>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.

# antenna.conf
# @output_html_templates = [
# ['template.html','index.html'],
# ]
#

require 'cgi'

def output_html
	templates = @output_html_templates || [ ['template.html','index.html'] ]

	update = Time.now.to_s

	templates.each do |i|
		next unless i && i[0] && i[1]
		template = File.expand_path( i[0], @confdir )
		output = File.expand_path( i[1], @confdir )
		next unless File.exist?( template )
		html = open( template ) do |f|
			f.read
		end
		html.gsub!( /%(title|copyright|antenna_url|rdf_url|update|generator)%/i ) do
			CGI::escapeHTML( eval( "defined?( #{$1} ) ? #{$1} : @#{$1}" ) )
		end
		html.gsub!( /%begin%\s*(.+)%end%\s*/im ) do
			line = $1
			r = ""
			@urls.sort {|a, b| urls_compare( a, b ) }.each do |item|
				next unless item
				name, author, url = item
				link = get_link( url )
				auth_url = @auth_url[url]||@antenna_url
				format = line.gsub( /%(name|author|url|link|auth_url)%/i ) do
					CGI::escapeHTML( eval( "defined?( #{$1} ) ? #{$1} : @#{$1}" ).gsub( /%/, '%%' ) )
				end
				if @last_modified[url] then
					r << @last_modified[url].strftime( format )
				else
					r << format.gsub( /%(.)/ ) do
						if $1 == '%' then
							'%'
						else
							'-' * Time.now.strftime( "%#{$1}" ).length
						end
					end
				end
			end
			r
		end

		open( output, "w" ) do |f|
			f.print html
		end
	end
end

add_output_proc do
	output_html
end


misc/plugin/output_lirs.rb
# output_lirs.rb $Revision: 1.3 $
#
# Copyright (C) 2004  Norihiro Hattori <tnh@webmasters.gr.jp>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.

# Last modified Information Relaying Specification ver.2.1
# http://aniki.haun.org/natsu/natsu3.1b/doc/LIRS.html

# antenna.conf
# @output_lirs_path = [
# 'natsumican.lirs',
# ]
#

def output_lirs
	lirs_paths = @output_lirs_path || ['rantenna.lirs']

	l = ""
	@urls.each do |item|
		next unless item
		l << <<-LIRS
LIRS,#{@last_modified[item[2]].to_i},#{@last_detected[item[2]].to_i},32400,#{@content_length[item[2]]||0},#{item[2]},#{item[0]},#{item[1]},#{@auth_url[item[2]]},#{item[3]},
LIRS
	end

	lirs_paths.each do |item|
		next unless item
		file = File.expand_path( item, @confdir )
		open( file,  "w" ) do |f|
			f.print l
		end
		begin
			require 'zlib'
			Zlib::GzipWriter.open( "#{file}.gz" ) do |gz|
				gz.write l
			end
		rescue LoadError
		end
	end
end

add_output_proc do
	output_lirs
end


misc/plugin/output_imode.rb
# output_imode.rb $Revision: 1.2 $
#
# Copyright (C) 2005  Michitaka Ohno <elpeo@mars.dti.ne.jp>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.

# antenna.conf
# @output_imode_path = [
# 'imode.html',
# ]
#

require 'cgi'

def output_imode
	paths = @output_imode_path || ['imode.html']

	title = CGI::escapeHTML( NKF::nkf( '-m0 -s', @title ) )
	html = <<-HTML
<HTML>
<HEAD>
<META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=Shift_JIS">
<TITLE>#{title}</TITLE>
</HEAD>
<BODY>
<H1>#{title}</H1>
HTML

	sorted_urls = @urls.sort {|a, b| urls_compare( a, b ) }
	sorted_urls.each_index do |i|
		next unless sorted_urls[i]
		name = CGI::escapeHTML( NKF::nkf( '-m0 -s', sorted_urls[i][0] ) )
		url = CGI::escapeHTML( NKF::nkf( '-m0 -s', sorted_urls[i][2] ) )
		sa = i < 10 ? [63879+i].pack( "n" ) : "\x81\xa0"
		ac = i < 10 ? %Q[ACCESSKEY="#{(i+1)%10}"] : ''
		if i == 10 then
			sa = "\xf9\x85"
			ac = 'ACCESSKEY="#"'
		end
		html << <<-HTML
<A HREF="#{url}" #{ac}>#{sa}#{name}</A><BR>
HTML
	end

	html << <<-HTML
</BODY>
</HTML>
HTML

	paths.each do |item|
		next unless item
		open( File.expand_path( item, @confdir ), "w" ) do |f|
			f.print html
		end
	end
end

add_output_proc do
	output_imode
end


go.rb
#!/usr/bin/env ruby
#
# go.rb $Revision: 1.4 $
#
# Copyright (C) 2004  Michitaka Ohno <elpeo@mars.dti.ne.jp>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.

# Inspired by a discussion of following URL:
# http://mput.dip.jp/mput/?date=20040316#p05

BEGIN { $defout.binmode }

require 'cgi'

if ENV['PATH_INFO'] then
	url = ENV['PATH_INFO'].split( /\//, 3 )[2].sub( /^http:\/+/, 'http://' )
	if ENV['QUERY_STRING'] && ENV['QUERY_STRING'] != "" then
		url << '?' + ENV['QUERY_STRING']
	end
else
	require File.expand_path( 'antenna.rb', File::dirname( __FILE__ ) )
	url = Antenna.new.antenna_url
end

print CGI.new.header( {"status" => "REDIRECT", "Location" => url} ) 

